apps/labs/posts/scipy-array-api.md

```diff
@@ -0,0 +1,441 @@
+
+NumPy is the n-dimensional array library which SciPy is built to work with - it is SciPy's only non-optional
+runtime dependency.
+At the time of writing, the vast majority of SciPy functions only work when providing NumPy arrays,
+despite the various interoperability efforts so far
+(see [Anirudh's blog post](https://labs.quansight.org/blog/array-libraries-interoperability)).
+The main problem with this is that "NumPy provides an array implementation that’s in-memory,
+CPU-only and single-threaded."
+The potential performance improvements that could be won with parallel algorithms, support for distributed
+arrays and support for GPUs (and other hardware accelerators) are huge, and thus "SciPy has had 'support
+for distributed and GPU arrays' on its roadmap" for five years now.
+See [this page](https://data-apis.org/array-api/latest/use_cases.html#use-case-scipy) on the use cases of the
+array API and [this blog post](https://labs.quansight.org/blog/2021/11/pydata-extensibility-vision) by
+Ivan Yashchuk and Ralf Gommers for more information.
+
+Due to these performance wishes, many different "array provider" libraries, such as CuPy and
+PyTorch, have been developed, with support for one or more of these hardware use-cases in mind.
+However, due to the fact that "array consumer" libraries are typically built to be compatible with
+only one array provider library, many functions have been reimplemented in different consumer libraries,
+just to work with a different provider library.
+In extreme cases, this has led to efforts to reimplement entire consumer libraries,
+for example [`cupyx.scipy`](https://docs.cupy.dev/en/stable/reference/scipy.html).
+This constant reimplementation of the same functionality has led many to see the Scientific Python Ecosystem
+as rather being fractured into many smaller ecosystems, one built around each array library, and to dream for a
+world where the ecosystems are unified and everything works together!
+
+While it would certainly be a very ambitious goal to aim for _everything_ working together, the goal
+of increasing interoperability between array libraries is one with widespread support.
+There are currently many hurdles which users can run into when navigating this fractured ecosystem.
+They can find that they have to convert between different array provider libraries in their projects, in order to
+make use of the full range of tools which have been developed.
+For users who decide to use a different provider library to take advantage of performance benefits, they can
+find themselves needing to learn new APIs to accomplish the same tasks, which are often similar but frustratingly
+different to those which they are used to.
+For large existing projects, this makes the prospect of switching to a different provider library a very daunting one.
+
+On the developer side of things, it seems like there is a lot of wasted time in reimplementing the same functionality
+for many different provider libraries.
```

Comment:
Might be helpful to add a note saying that SciPy and other such libraries are still very much developer by volunteers. This can be linked to the above note saying that it was for 5 years on the roadmap. i.e. a start of an explanation of why now.

Reply:
Something like this?
```suggestion
On the developer side of things, it seems like there is a lot of wasted time in reimplementing the same functionality
for many different provider libraries.
Developer time is very valuable since these projects are still largely maintained by volunteers (and hence large
tasks like adding GPU support can stay on the roadmap for many years), so work to reduce the overall maintenance
load is key to accelerating progress.
```